### PR TITLE
Add thread safety macro CMS_SA_ALLOW == [[cms::sa_allow]]

### PR DESCRIPTION
**Pull request by gartung**
CMS_SA_ALLOW will be put in front of statements that the clang static analyzer module should skip.

[[cms::sa_allow]]  attribute will be added to clang in an external PR.

### CODE REVIEW

File: FWCore/Framework/interface/Event.h [hunk 1]
```diff
@@ -306,23 +307,23 @@ namespace edm {
     ProductPtrVec putProducts_;
 
     EventAuxiliary const& aux_;
-    mutable std::optional<LuminosityBlock> luminosityBlock_;
+    CMS_SA_ALLOW mutable std::optional<LuminosityBlock> luminosityBlock_;
```
Comment (makortel) [inline FWCore/Framework/interface/Event.h]: Now we should also add the explanation why the `mutable` is allowed  :)

Comment (gartung) [reply FWCore/Framework/interface/Event.h]: What is the comment you would like added. I will add it to the PR.

Comment (Dr15Jones) [reply FWCore/Framework/interface/Event.h]:
```
    //measurable performance gain by only creating LuminosityBlock when needed
    CMS_SA_ALLOW mutable std::optional<LuminosityBlock> luminosityBlock_;
```

**Review by Dr15Jones — CHANGES_REQUESTED** [review]

File: FWCore/Framework/interface/Event.h (hunk 1 above)
Comment (makortel) [reply FWCore/Framework/interface/Event.h]: Should we add something about `edm::Event` object being intended to be accessed only from one thread?

**Comment by Dr15Jones** [comment]
please test

**Comment by gartung** [comment]
abort

**Comment by gartung** [comment]
please test with  cms-externals/llvm-project#3

**Comment by Dr15Jones** [comment]
+1

**Comment by fabiocos** [comment]
+1

**Comment by Dr15Jones** [comment]
The most recent changes to our LLVM external did the trick and this is now working.

**Pull request merged**